### DISCUSSION TITLE
Confusion in docs, Can we use Face Models in NodeJS for finding similarities and matches ?

### DISCUSSION THREAD

**Discussion by uzair004** [vladmandic/human#137] [Q&A] [answered]
### Can we use this in the NodeJS backend for finding face matches and similarities?

I found docs a little confusing as I want to build a system that will check an image match or similarity against all images I have in the database. It is mentioned twice in docs that face models are not supported in NodeJS as of now, but still, they are used in NodeJS demo, can I use face models to create face image search and identity system?

**Screenshots of docs:**

<img width="782" alt="node human" src="https://user-images.githubusercontent.com/34101864/126056880-3347ff42-2b3f-40bb-8bf6-83b309b15e7f.PNG">

<img width="704" alt="node human 2" src="https://user-images.githubusercontent.com/34101864/126056885-442ea896-8233-48c0-a7e5-fc59eff1b0d1.PNG">

**Comment by vladmandic** [comment, marked as answer]
I'll update docs when I'm back in ~10 days.
This used to be true, now all models work in `NodeJS` and only missing functionality is advanced face rotation correction.

The required function for that last piece of functionality was just implemented in `tfjs-node` package so that should also be enabled in the next version of Human in about ~2 weeks.

**Reply by uzair004** [reply]
Please consider updating docs for this as people may leave the repo considering it doesn't support face models.

**Reply by vladmandic** [reply]
Like i Said - i Will update docs as soon as I'm back from my trip.

**Reply by neckaros** [reply]
As it been implemented?
i see in the doc there is still:
https://github.com/vladmandic/human/blob/497ca46006c1f57623c5c859de9f8b620e860ff8/TODO.md?plain=1#L41

> Enhanced rotation correction for face detection and hand detection is not working in **NodeJS** due to missing kernel op in **TFJS**  

That means that everything in the doc works for node JS expect precise face angle?

> Crop to face ✅
> Find rought face angle and straighten face  ✅
> Detect mesh  ✅
> Find precise face angle and again straighten face ❌
> Crop again with more narrow margins  ✅
> Convert image to grayscale to avoid impact of different colorizations  ✅
> Normalize brightness to full range for all images ✅

**Reply by vladmandic** [reply]
Your checklist is correct.

**Comment by vladmandic** [comment]
docs are updated. note that latest list of open issues is always listed in <TODO.md>

**Reply by uzair004** [reply]
It looks like it is not updated here [Platform support](https://github.com/vladmandic/human/wiki/Platforms#nodejs)

**Reply by vladmandic** [reply]
oversight on my part, corrected - thanks for reporting it.
